qiskit/algorithms/minimum_eigensolvers/sampling_mes.py

```diff
@@ -0,0 +1,128 @@
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> "SamplingMinimumEigensolverResult":
+        """Compute the minimum eigenvalue of a diagonal operator.
+
+        Args:
+            operator: Diagonal qubit operator.
+            aux_operators: Optional list of auxiliary operators to be evaluated with the
+                final state.
+
+        Returns:
+            A :class:`~.SamplingMinimumEigensolverResult` containing the optimization result.
+        """
+        pass
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        """Whether computing the expectation value of auxiliary operators is supported.
+
+        If the minimum eigensolver computes an eigenstate of the main operator then it
+        can compute the expectation value of the aux_operators for that state. Otherwise
+        they will be ignored.
+
+        Returns:
+            True if aux_operator expectations can be evaluated, False otherwise
+        """
+        return False
+
+
+class SamplingMinimumEigensolverResult(AlgorithmResult):
+    """Sampling Minimum Eigensolver Result.
+
+    In contrast to the result of a :class:`~.MinimumEigenSolver`, this result also contains
+    the best measurement of the overall optimization and the samples of the final state.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._eigenvalue = None
+        self._eigenstate = None
+        self._aux_operator_values = None
+        self._best_measurement = None
+
+    @property
+    def eigenvalue(self) -> complex | None:
+        """Return the approximation to the eigenvalue."""
+        return self._eigenvalue
+
+    @eigenvalue.setter
+    def eigenvalue(self, value: complex | None) -> None:
+        """Set the approximation to the eigenvalue."""
+        self._eigenvalue = value
+
+    @property
+    def eigenstate(self) -> Mapping[int, float] | None:
+        """Return samples of the final state."""
+        return self._eigenstate
+
+    @eigenstate.setter
+    def eigenstate(self, value: Mapping[int, float] | None) -> None:
+        """Set samples of the final state."""
+        self._eigenstate = value
+
+    @property
+    def aux_operator_values(self) -> ListOrDict[tuple[complex, complex]] | None:
+        """Return aux operator expectation values.
+
+        These values are in fact tuples formatted as (mean, standard deviation).
+        """
+        return self._aux_operator_values
+
+    @aux_operator_values.setter
+    def aux_operator_values(self, value: ListOrDict[tuple[complex, complex]] | None) -> None:
+        """set aux operator eigen values"""
+        self._aux_operator_values = value
+
+    @property
+    def best_measurement(self) -> Mapping[str, Any] | None:
+        """Return the best measurement (as bitstring) over the entire optimization."""
+        return self._best_measurement
```

Comment:
The description and type do not really seem to align for me. Is this a measurement Mapping as per the DiagonalEstimator above - with the Mapping[str, Any] it looks like that. We may want to state the fields in this dict for the user if this is the case,